### ISSUE TITLE
Launching a VS Code workspace via this extension is not the same as launching from command line

### ISSUE THREAD

**Issue by joekaiser** [Kazumasan/ulauncher-sessions#2]
Same issue in the project https://github.com/brpaz/ulauncher-vscode-projects/issues/11

When launching a workspace via this extension the $PATH variable is not fully loaded. The exact error I get is "Can not find npm on PATH" when VS code tries to run a script in the pacakge.JSON.

When you open the same workspace file via the command line `code projectName.code-workspace` Everything works fine.

**Comment by Kazumasan** [comment]
I've looked at the issue and if you have `isCLI: true` in your actions it sets `shell=True` for `subprocess.Popen()`. But i look into it further. If you have a solution feel free to make a PR.

**Comment by joekaiser** [comment]
I have been looking into it, but I can't figure it out. (Granted, I only know a little about Python.) 

It behaves like the process is open in some kind of sub-shell that doesn't have my `~/.profile` loaded

I tried forcing the ENV over [like described here,](https://stackoverflow.com/questions/2231227/python-subprocess-popen-with-a-modified-environment) but this didn't help.

```
def execAction(cmd_path, opt):
    my_env = os.environ.copy()
    subprocess.Popen(
        cmd_path + " " + opt[0], shell=opt[1]["isCLI"], cwd=os.path.dirname(opt[0]),
        env=my_env
    )
```

Maybe it is a python bug?

**Comment by Kazumasan** [comment]
@joekaiser can you try running it in the "Dev" mode with `ulauncher --noextension --dev` and then run the extension seperatly with the command that is given in the extension tab.

 if that works  I'm afraid it might have something to do with how ulauncher runs the extension.

Edit: I'm still unable to understand the issue exactly i made a test directory and i think what you mean is that if click start under npm scripts in vs code the integrated terminal will run "npm start" and that is causing problems ? and relaunching ulauncher might also have an effect already.
![Screenshot_20210314_131823](https://user-images.githubusercontent.com/19864466/111068143-d6152380-84c7-11eb-8f42-bd458133ddee.png)

**Comment by joekaiser** [comment]
**edit** I have NPM installed via NVM

I'm sorry I wasn't clear - the integrated terminal is fine.

- Create a node project. Any basic hello world is fine.
- Switch to the Debugging tab and create a basic launch script to run the Start script.
- Run the new launch definition you made and you will get the error. If you close Code and open it a traditional way the script is able to run 

![image](https://user-images.githubusercontent.com/1521110/111071780-faa3d800-8494-11eb-8757-322b955891ee.png)


`launch.json`
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "type": "node",
      "request": "launch",
      "name": "Launch via NPM",
      "runtimeExecutable": "npm",
      "runtimeArgs": [
        "run-script",
        "start"
      ],
      "port": 65000,
      "outputCapture": "std"
    }
  ]
}
```
![image](https://user-images.githubusercontent.com/1521110/111071797-08595d80-8495-11eb-81f4-e51dc5c01534.png)


![image](https://user-images.githubusercontent.com/1521110/111071801-0b544e00-8495-11eb-968b-efd159354537.png)

**Comment by joekaiser** [comment]
Running in dev mode fixes the problem. Does that mean it is how uLauncher runs the extension?

**Comment by Kazumasan** [comment]
it might be. try relaunching ulauncher from the terminal and from your app launcher just don't let autostart do it. It might be that autostart launches application with different vars / before `~/.profile` is sourced.

**Comment by joekaiser** [comment]
Launching it from the terminal works, but via my app launcher it fails.

**Comment by Kazumasan** [comment]
i thought so... so env will not be provided by whatever launches ulauncher for you. And i don't have this issue just tested it.

but i keep looking into it to find a way to provide env with that but thats probably why `my_env = os.environ.copy()` didn't work because it provided falsey env variables.

maybe running ulauncher form `~./profile` as a dirty fix ?

edit: i cant totally rule out NVM though it should not interfere.

**Comment by joekaiser** [comment]
Yeah, I was thinking the same thing. It isn't great, but it does solve the problem.

thank you :)

**Comment by joekaiser** [comment]
So I think the real problem is that when the app launcher starts a program, it uses a non-interactive shell.

Remember I said I used NVM to install node? My PATH is set to include NVM_DIR in my .zshrc **not* .profile. As soon as I added this to `~/.profile` everything works fine.  You probably have to logout first.
```
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
```

Well, fine is relative :)  NPM is now found but none of my env vars are that my app needs. I think we have figured out this isn't a bug in this extension, so I will close it.

Thank you so much for your help. :green_heart:
